### ISSUE TITLE
switch_rtp.c:3277 audio Handshake failure 1.

### ISSUE THREAD

**Issue by yzh1127116449** [signalwire/freeswitch#1249]
I use freeswitch-1.10.6-release on CentOS, get an error.  Cause I get through and hang up
like this:
![image](https://user-images.githubusercontent.com/30822672/124012823-29c66100-da14-11eb-8d46-dd8f85250138.png)
log :
2021-07-01 01:53:39.543993 [DEBUG] switch_ivr_originate.c:3883 Originate Resulted in Success: [sofia/internal/8rhg155t@192.0.2.112]
2021-07-01 01:53:39.543993 [DEBUG] switch_ivr_bridge.c:1793 (sofia/internal/8rhg155t@192.0.2.112) State Change CS_CONSUME_MEDIA -> CS_EXCHANGE_MEDIA
2021-07-01 01:53:39.543993 [DEBUG] switch_core_state_machine.c:585 (sofia/internal/8rhg155t@192.0.2.112) Running State Change CS_EXCHANGE_MEDIA (Cur 2 Tot 2)
2021-07-01 01:53:39.543993 [DEBUG] switch_core_state_machine.c:654 (sofia/internal/8rhg155t@192.0.2.112) State EXCHANGE_MEDIA
2021-07-01 01:53:39.543993 [DEBUG] mod_sofia.c:656 SOFIA EXCHANGE_MEDIA
2021-07-01 01:53:39.623986 [WARNING] switch_rtp.c:3277 audio Handshake failure 1. This may happen when you use legacy DTLS v1.0 (legacyDTLS channel var is set) but endpoint requires DTLS v1.2.
2021-07-01 01:53:39.623986 [INFO] switch_rtp.c:3278 Changing audio DTLS state from HANDSHAKE to FAIL
2021-07-01 01:53:39.663987 [NOTICE] switch_rtp.c:3259 Hangup sofia/internal/8rhg155t@192.0.2.112 [CS_EXCHANGE_MEDIA] [DESTINATION_OUT_OF_ORDER]
2021-07-01 01:53:39.663987 [DEBUG] switch_ivr_bridge.c:912 BRIDGE THREAD DONE [sofia/internal/8rhg155t@192.0.2.112]
2021-07-01 01:53:39.663987 [DEBUG] switch_core_state_machine.c:654 (sofia/internal/8rhg155t@192.0.2.112) State EXCHANGE_MEDIA going to sleep
2021-07-01 01:53:39.663987 [DEBUG] switch_core_state_machine.c:585 (sofia/internal/8rhg155t@192.0.2.112) Running State Change CS_HANGUP (Cur 2 Tot 2)
2021-07-01 01:53:39.663987 [DEBUG] switch_core_state_machine.c:848 (sofia/internal/8rhg155t@192.0.2.112) Callstate Change ACTIVE -> HANGUP
2021-07-01 01:53:39.663987 [DEBUG] switch_core_state_machine.c:850 (sofia/internal/8rhg155t@192.0.2.112) State HANGUP
2021-07-01 01:53:39.663987 [DEBUG] mod_sofia.c:453 Channel sofia/internal/8rhg155t@192.0.2.112 hanging up, cause: DESTINATION_OUT_OF_ORDER
2021-07-01 01:53:39.683985 [DEBUG] mod_sofia.c:507 Sending BYE to sofia/internal/8rhg155t@192.0.2.112
2021-07-01 01:53:39.683985 [DEBUG] switch_core_state_machine.c:60 sofia/internal/8rhg155t@192.0.2.112 Standard HANGUP, cause: DESTINATION_OUT_OF_ORDER
2021-07-01 01:53:39.683985 [DEBUG] switch_core_state_machine.c:850 (sofia/internal/8rhg155t@192.0.2.112) State HANGUP going to sleep
2021-07-01 01:53:39.683985 [DEBUG] switch_core_state_machine.c:620 (sofia/internal/8rhg155t@192.0.2.112) State Change CS_HANGUP -> CS_REPORTING
2021-07-01 01:53:39.683985 [DEBUG] switch_core_state_machine.c:585 (sofia/internal/8rhg155t@192.0.2.112) Running State Change CS_REPORTING (Cur 2 Tot 2)
send 596 bytes to wss/[192.168.75.12]:59411 at 01:53:39.684423:
------------------------------------------------------------------------
BYE sip:8rhg155t@192.0.2.112;transport=wss SIP/2.0

what I did:
1.I have update openssl version to  OpenSSL 1.0.2k  26 Jan 2017
![image](https://user-images.githubusercontent.com/30822672/124015167-f46f4280-da16-11eb-9726-c208421995e6.png)
2.recompile freeswitch 
3.delete the file /certs/dtls-srtp.pem 
4.start freeswitch. The exception is still there

freeswitch - Version [1.10.6]
chrome -version [91.0.4472.124]

My usage scenario:
1.Two registered users,One is registered through WSS（100001,The other is a normal SIP registry（100002），then，100002 call 100001
2. xml config:
3. 
![image](https://user-images.githubusercontent.com/30822672/124015076-de618200-da16-11eb-9133-4c3bbb1cb69f.png)

     <extension name="callin">
      <condition field="destination_number" expression="^(1\d{5})$">
        <action application="export" data="media_webrtc=true"/>
        <action application="bridge"  data="user/$1@${domain_name}"/>
      </condition>
    </extension>

How do I solve this problem,thanks for help!

**Comment by dragos-oancea** [comment]
https://lists.freeswitch.org/pipermail/freeswitch-users/2019-December/132869.html

**Comment by yzh1127116449** [comment]
hi, all
&nbsp; &nbsp;thank you for your reply, but&nbsp;I can't open this website,&nbsp;Are you sure this website is accessible?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "signalwire/freeswitch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月1日(星期四) 下午5:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [signalwire/freeswitch] switch_rtp.c:3277 audio Handshake failure 1. (#1249)





 
https://lists.freeswitch.org/pipermail/freeswitch-users/2019-December/132869.html

**Comment by dragos-oancea** [comment]
yes, it is. url says: 
```Yes, thanks, the issue was Centos 7.x does not uspport the correct version,
I had to move to Centos 8.

><>
nathan stratton


On Sun, Dec 29, 2019 at 6:47 PM Piotr Gregor <piotr at dataandsignal.com>
wrote:

> Hi Nathan,
>
> The best option is to upgrade system to one that ships with SSL above
> 1.1.0, like for instance new Debian 10 "Buster".
>
> peter at photon:~/$ openssl version
> OpenSSL 1.1.1d  10 Sep 2019
>
> cheers,
>
>
>
>```

**Comment by yzh1127116449** [comment]
hi&nbsp;Dragos Oancea
&nbsp; &nbsp; Thank you very much. I do that you say ,Update OpenSSL to 1.1.1d, but com&nbsp; piler Freeswitch 1.10.6 returns an error:
&nbsp; &nbsp; &nbsp; error info :&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; &nbsp;then I&nbsp; clone the master branch code , complile successed. and&nbsp; verify that my problem has been solved.&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp;thanks&nbsp;for&nbsp;your&nbsp;help, wish you a happy life, smooth work




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "signalwire/freeswitch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月1日(星期四) 晚上8:39
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [signalwire/freeswitch] switch_rtp.c:3277 audio Handshake failure 1. (#1249)





 
yes, it is. url says:
 I had to move to Centos 8. &gt;<&gt; nathan stratton On Sun, Dec 29, 2019 at 6:47 PM Piotr Gregor <piotr at dataandsignal.com&gt; wrote: &gt; Hi Nathan, &gt; &gt; The best option is to upgrade system to one that ships with SSL above &gt; 1.1.0, like for instance new Debian 10 "Buster". &gt; &gt; peter at photon:~/$ openssl version &gt; OpenSSL 1.1.1d  10 Sep 2019 &gt; &gt; cheers, &gt; &gt; &gt; &gt;```

**Comment by yzh1127116449** [comment]
hi Dragos Oancea

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; I have a problem about webrtc, thanks for your help. thank you very much!

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 1. User 100001 has registered with Freeswitch(Freeswitch is running on a public network) through the WSS.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 2. Run the fs_CLI command to connect to Freeswitch

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 3. run :originate user/internal/100002 &amp;echo

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4. When I dialed, I couldn't hear anything

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 5. the sdp info, ip is local address , I do not know how to change the public IP address.



***@***.***&gt; global_getvar
hostname=znwh-ahdx-cdsz-01
local_ip_v4=192.168.75.7
local_mask_v4=255.255.255.224
local_ip_v6=::1
base_dir=/usr/local/freeswitch
recordings_dir=/usr/local/freeswitch/recordings
sounds_dir=/usr/local/freeswitch/sounds
conf_dir=/usr/local/freeswitch/conf
log_dir=/usr/local/freeswitch/log
run_dir=/usr/local/freeswitch/run
db_dir=/usr/local/freeswitch/db
mod_dir=/usr/local/freeswitch/mod
htdocs_dir=/usr/local/freeswitch/htdocs
script_dir=/usr/local/freeswitch/scripts
temp_dir=/tmp
grammar_dir=/usr/local/freeswitch/grammar
fonts_dir=/usr/local/freeswitch/fonts
images_dir=/usr/local/freeswitch/images
certs_dir=/usr/local/freeswitch/certs
storage_dir=/usr/local/freeswitch/storage
cache_dir=/usr/local/freeswitch/cache
data_dir=/usr/local/freeswitch
localstate_dir=/usr/local/freeswitch
default_password=aikf1234
sound_prefix=/usr/local/freeswitch/sounds/en/us/callie
domain=192.168.75.7
domain_name=192.168.75.7
hold_music=local_stream://moh
use_profile=external
rtp_sdes_suites=AEAD_AES_256_GCM_8|AEAD_AES_128_GCM_8|AES_CM_256_HMAC_SHA1_80|AES_CM_192_HMAC_SHA1_80|AES_CM_128_HMAC_SHA1_80|AES_CM_256_HMAC_SHA1_32|AES_CM_192_HMAC_SHA1_32|AES_CM_128_HMAC_SHA1_32|AES_CM_128_NULL_AUTH
zrtp_secure_media=true
global_codec_prefs=PCMU,PCMA
outbound_codec_prefs=PCMU,PCMA
xmpp_client_profile=xmppc
xmpp_server_profile=xmpps
bind_server_ip=auto
external_rtp_ip=8.142.67.180
external_sip_ip=8.142.67.180
unroll_loops=true
outbound_caller_name=FreeSWITCH
outbound_caller_id=0000000000
call_debug=false
console_loglevel=info
default_areacode=918
default_country=US
presence_privacy=false
au-ring=%(400,200,383,417);%(400,2000,383,417)
be-ring=%(1000,3000,425)
ca-ring=%(2000,4000,440,480)
cn-ring=%(1000,4000,450)
cy-ring=%(1500,3000,425)
cz-ring=%(1000,4000,425)
de-ring=%(1000,4000,425)
dk-ring=%(1000,4000,425)
dz-ring=%(1500,3500,425)
eg-ring=%(2000,1000,475,375)
es-ring=%(1500,3000,425)
fi-ring=%(1000,4000,425)
fr-ring=%(1500,3500,440)
hk-ring=%(400,200,440,480);%(400,3000,440,480)
hu-ring=%(1250,3750,425)
il-ring=%(1000,3000,400)
in-ring=%(400,200,425,375);%(400,2000,425,375)
jp-ring=%(1000,2000,420,380)
ko-ring=%(1000,2000,440,480)
pk-ring=%(1000,2000,400)
pl-ring=%(1000,4000,425)
ro-ring=%(1850,4150,475,425)
rs-ring=%(1000,4000,425)
ru-ring=%(800,3200,425)
sa-ring=%(1200,4600,425)
tr-ring=%(2000,4000,450)
uk-ring=%(400,200,400,450);%(400,2000,400,450)
us-ring=%(2000,4000,440,480)
bong-ring=v=-7;%(100,0,941.0,1477.0);v=-7;&gt;=2;+=.1;%(1400,0,350,440)
beep=%(1000,0,640)
sit=%(274,0,913.8);%(274,0,1370.6);%(380,0,1776.7)
df_us_ssn=(?!219099999|078051120)(?!666|000|9\d{2})\d{3}(?!00)\d{2}(?!0{4})\d{4}
df_luhn=?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|6(?:011|5[0-9]{2})[0-9]{12}|(?:2131|1800|35\d{3})\d{11}
default_provider=example.com
default_provider_username=joeuser
default_provider_password=password
default_provider_from_domain=example.com
default_provider_register=false
default_provider_contact=5000
sip_tls_version=tlsv1.2,sslv2,sslv23,sslv3
***@***.***
internal_auth_calls=true
internal_sip_port=5060
internal_tls_port=5061
internal_ssl_enable=true
external_auth_calls=true
external_sip_port=5080
external_tls_port=5081
external_ssl_enable=true
rtp_video_max_bandwidth_in=3mb
rtp_video_max_bandwidth_out=3mb
suppress_cng=true
rtp_liberal_dtmf=true
video_mute_png=/usr/local/freeswitch/images/default-mute.png
video_no_avatar_png=/usr/local/freeswitch/images/default-avatar.png
aikf_sip_ip=192.168.75.7
aikf_rtp_ip=192.168.75.7
aikf_ext_sip_ip=192.168.75.7
aikf_ext_rtp_ip=192.168.75.7
aikf_gateway=192.168.75.7:5090
AT_EPENT1=0 0 0 -1 -1 0 -1 0 -1 -1 0 -1
AT_EPENT2=1 1 1 -1 -1 1 -1 1 -1 -1 1 -1
AT_CPENT1=0 -1 -1 0 -1 0 0 0 -1 -1 0 -1
AT_CPENT2=1 -1 -1 1 -1 1 1 1 -1 -1 1 -1
AT_CMAJ1=0 -1 0 0 -1 0 -1 0 0 -1 0 -1
AT_CMAJ2=1 -1 1 1 -1 1 -1 1 1 -1 1 -1
AT_BBLUES=1 -1 1 -1 -1 1 -1 1 1 1 -1 -1
ATGPENT2=-1 1 -1 1 -1 1 -1 -1 1 -1 1 -1
core_uuid=4004cbc6-061f-4fbf-aa44-38a5ecb55d5c





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "signalwire/freeswitch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月6日(星期二) 下午3:55
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [signalwire/freeswitch] switch_rtp.c:3277 audio Handshake failure 1. (#1249)





 
Closed #1249.

**Comment by dragos-oancea** [comment]
maybe use uuid_displace to play something.
